src/pineko/check.py

```diff
@@ -111,43 +111,38 @@ def check_grid_and_eko_compatible(pineappl_grid, operators, xif, max_as, max_al)
         raise ValueError("x grid in pineappl grid and eko operator are NOT compatible!")
 
 
-def is_dis(channels):
+def is_dis(convolutions):
     """Check if the fktable we are computing is a DIS fktable.
 
     Parameters
     ----------
-    channels : list(list(tuple(list, float)))
-        object containing the information on the channels. From PineAPPL v1,
-        the tuple contains two elements. The first is a list containing the
-        PIDs of the partons. The length of the list coincides with the number
-        of partons involved, ie for DIS there is only one element, and for a
-        hadron production in pp collision there are three elements. The 2nd
-        element is the weight/multiplicative factor.
+    convolutions: pineappl.grid.convolutions
+           an object containing the information on the convolution
 
     Returns
     -------
     bool
         true if the fktable is a DIS fktable
     """
-    return True if len(channels[0][0][0]) == 1 else False
+    return True if len(convolutions) == 1 else False
```

Comment:
you also need to check if it is space-like - else it might be also SIA ...

Reply:
But SIA always involves two convolution though.

Reply:
Ah, you are thinking of these processes. Because we won't have measurements and predictions for these in the foreseeable future, I didn't think about them. But you're right, we should also account for these cases. I'll add such modifications.

Reply:
hihi, nono - SIA = semi-inclusive annihilation and this is the baseline for all current FF extractions. We have lot's of data: look e.g. to Table 1 of [2503.21311](https://arxiv.org/pdf/2503.21311) . Also Belle is still actively taking data I think

Reply:
hahaha, of course. I was being very stupid. I still did have in mind SIDI while still writing SIA. My brain is not functioning well because of exhaustion 🙈